### PR TITLE
JetBrains: display visual hint about cody autocomplete status in the status bar

### PR DESCRIPTION
**Pull request by szymonprz**
Previously there were no information about the status of autocomplete progress from Cody. Now we display the Cody icon in the status bar that changes depending on the actions that Cody is doing. 
There are four possible states of the Cody in the status bar:
- Cody is disabled -> icon is not shown
- Cody is enabled but autocomplete is disabled - muted cody icon is shown
- Cody autocomplete is enabled - standard cody icon is shown
- Autocomplete processing is in progress - animated spinning loading is shown

Icons displayed in the status bar respect selected theme.  

Screenshot 2023-08-21 at 12 46 27

closes https://github.com/sourcegraph/sourcegraph/issues/55909

## Test plan
- Change Cody configuration to disable it/enable it, disable and enable autocomplete, the icons in the status bar should change
- Spinning loader icon should appear when Cody is processing the autocomplete items

### CODE REVIEW

File: client/jetbrains/src/main/kotlin/com/sourcegraph/cody/statusbar/CodyAutocompleteStatusService.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    if (notify) {
+      updateCodyStatusBarIcons()
+    }
+  }
+
+  private fun updateCodyStatusBarIcons() {
+    val action = Runnable {
+      val openProjects = ProjectManager.getInstance().openProjects
+      openProjects.forEach { it.takeIf { !it.isDisposed }?.let { CodyStatusBarWidget.update(it) } }
```
Comment (olafurpg) [inline client/jetbrains/src/main/kotlin/com/sourcegraph/cody/statusbar/CodyAutocompleteStatusService.kt:34]: nit: this expression would be clearer to read if it used less Kotlin sugar

```
openProjects.foreach { project ->
  if (!project.isDisposed) {
    CodyStatusBarWidget.update(project)
  }
}
```

File: client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/CodyAutoCompleteManager.java [hunk 2]
```diff
@@ -159,54 +161,71 @@ private CompletableFuture<Void> triggerAutoCompleteAsync(
...
-            }
-            return;
-          }
-          final InlineAutoCompleteItem item = maybeItem.get();
-          try {
-            ApplicationManager.getApplication()
-                .invokeLater(
-                    () -> {
-                      this.clearAutoCompleteSuggestions(editor);
+    return CompletableFuture.runAsync(
```
Comment (olafurpg) [inline client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/CodyAutoCompleteManager.java]: What happens if we do a blocking `CodyAutocompleteStatusService.notifyApplication(CodyAutocompleteStatus.AutocompleteInProgress)` call?

I'm not 100% sure, but I think this diff changes where the work happens. Previously, we used the lsp4j pool while now 
`CompletableFuture.runAsync` runs in `ForkJoinPool.commonPool()`. This is most likely not a problem but I ideally want to limit the number of thread pools where Cody runs.

File: client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/CodyAutoCompleteManager.java [hunk 3]
```diff
@@ -159,54 +161,71 @@ private CompletableFuture<Void> triggerAutoCompleteAsync(
...
+            () ->
+                CodyAutocompleteStatusService.notifyApplication(
+                    CodyAutocompleteStatus.AutocompleteInProgress))
+        .thenCompose(
+            unused ->
```
Comment (vdavid) [inline client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/CodyAutoCompleteManager.java]: Is `unused` idiomatic Java? I used `_` before, but maybe that's only idiomatic in Go.

**Review by olafurpg — APPROVED** [review]
LGTM 👍🏻 This is a great improvement. I pushed a commit to avoid `CompletableFuture.runAsync`

File: client/jetbrains/src/main/kotlin/com/sourcegraph/cody/statusbar/CodyAutocompleteStatus.kt [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+    override val icon: Icon = Icons.StatusBar.CodyAutocompleteDisabled
+  },
+  Ready {
+    override fun getPresentableText(): String = "Cody autocomplete is enabled"
+
+    override val icon: Icon = Icons.StatusBar.CodyAvailable
+  },
+  AutocompleteInProgress {
+    override fun getPresentableText(): String = "Cody autocomplete is in progress"
```
Comment (vdavid) [inline client/jetbrains/src/main/kotlin/com/sourcegraph/cody/statusbar/CodyAutocompleteStatus.kt:28]: Nit: I'd add the three dots. Whichever you prefer.
```suggest
    override fun getPresentableText(): String = "Cody autocomplete is in progress..."
```

File: client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/render/CodyAutoCompleteElementRenderer.java [hunk 5]
```diff
@@ -26,6 +26,8 @@ public CodyAutoCompleteElementRenderer(
       @Nullable AutoCompleteRendererType type) {
     this.text = text;
     this.completionItem = completionItem;
+    int a = 42;
```
Comment (vdavid) [inline client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/render/CodyAutoCompleteElementRenderer.java:29]: I think this is not related to the PR :D

**Review by vdavid — APPROVED** [review]
Generally looks good, left 3 comments

**Comment by vdavid** [comment]
@olafurpg I disabled auto-merge because I think there are some accidental logs from your last commit

File: client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/render/CodyAutoCompleteElementRenderer.java (hunk 5 above)
Comment (olafurpg) [reply client/jetbrains/src/main/java/com/sourcegraph/cody/autocomplete/render/CodyAutoCompleteElementRenderer.java]: Whoops! Fixed now. Good catch

**Comment by olafurpg** [comment]
I opened a followup issue to add the ability to click on the icon to enable/disable autocomplete https://github.com/sourcegraph/sourcegraph/issues/56105
